src/EditorFeatures/Core/Implementation/Diagnostics/AbstractDiagnosticsTaggerProvider.AggregatingTagger.cs

```diff
@@ -174,10 +174,6 @@ private void OnDiagnosticsUpdated(object sender, DiagnosticsUpdatedArgs e)
 
             private void OnDiagnosticsUpdated(DiagnosticsUpdatedArgs e)
             {
-                // First see if this is a document/project removal.  If so, clear out any state we
-                // have associated with any analyzers we have for that document/project.
-                ProcessRemovedDiagnostics(e);
```

Comment:
I can't really fix it because I don't understand the contract of the diagnostic services**. That's why I'm asking for those who do understand to make an appropriate API that can be called so that we won't have race conditions and other issues that will be hard to track down. 

** The existing APIs are difficult enough to understand that I have no ability to review this change out even determine if it does the right thing. That's not good